src/main.js

```diff
@@ -277,13 +279,79 @@ let load = loader({
     },
   },
 
+  s3Controller: {
+    requires: ['cfg'],
+    setup: async ({cfg}) => {
+      return new remoteS3.Controller({
+        region: cfg.app.blobArtifactRegion,
+        accessKeyId: cfg.aws.accessKeyId,
+        secretAccessKey: cfg.aws.secretAccessKey,
+      });
+    },
+  },
+
+  // This is a dependency to ease the creation of buckets for, e.g. testing
+  createBuckets: {
+    requires: ['cfg'],
+    setup: async ({cfg}) => {
+      let s3 = new aws.S3({
+        region: cfg.app.blobArtifactRegion,
+        accessKeyId: cfg.aws.accessKeyId,
+        secretAccessKey: cfg.aws.secretAccessKey,
+      });
+
+      // First, we'll create a public bucket
+      let publicConfig = {
+        Bucket: cfg.app.publicBlobArtifactBucket,
+        ACL: 'public-read',
+      };
+      
+      // Next, the private one
+      let privateConfig = {
+        Bucket: cfg.app.privateBlobArtifactBucket,
+        ACL: 'private',
+      };
+
+      if (cfg.app.blobArtifactRegion !== 'us-east-1') {
+        let x = (y) => {
+          y.CreateBucketConfiguration = {
+            LocationConstraint: cfg.app.blobArtifactRegion
+          };
+        }
+        x(publicConfig);
+        x(privateConfig);
```

Comment:
if you really want to do this, use a name that is understandable like `config` instead of `x`..
Or do something like: `[publicConfig, privateConfig].map(config => ....)` but really, I think duplication is the simplest answer here we only have two.